eduVPN/AuthenticationService.swift

```diff
@@ -80,8 +80,16 @@ class AuthenticationService {
         let configuration = OIDServiceConfiguration(authorizationEndpoint: info.authorizationURL, tokenEndpoint: info.tokenURL)
         
         redirectHTTPHandler = OIDRedirectHTTPHandler(successURL: nil)
-        let redirectURL = URL(string: "callback", relativeTo: redirectHTTPHandler!.startHTTPListener(nil))!
-        let request = OIDAuthorizationRequest(configuration: configuration, clientId: "org.eduvpn.app.macos", clientSecret: nil, scopes: ["config"], redirectURL: redirectURL, responseType: OIDResponseTypeCode, additionalParameters: nil)
+        var redirectURL: URL?
+        if Thread.isMainThread {
+            redirectURL = redirectHTTPHandler!.startHTTPListener(nil)
+        } else {
+            DispatchQueue.main.sync {
```

Comment:
The problem was with OIDRedirectHTTPHandler requiring running CFRunLoop (on a thread, where startHTTPListener is called) to handle HTTP requests. GCD worker thread doesn't have one, so here we just dispatching it to main thread.